ledger/txtail.go

```diff
@@ -213,6 +213,9 @@ func (t *txTail) newBlock(blk bookkeeping.Block, delta ledgercore.StateDelta) {
 
 	t.tailMu.Lock()
 	defer t.tailMu.Unlock()
+	if delta.Txleases == nil {
+		delta.Txleases = make(map[ledgercore.Txlease]basics.Round)
+	}
 	t.recent[rnd] = roundLeases{
 		txleases: delta.Txleases,
 		proto:    config.Consensus[blk.CurrentProtocol],
```

Comment:
I think we should not allocate into `delta.Txleases` but into a local var instead:
```suggestion
	leases := delta.Txleases
	if delta.Txleases == nil {
		leases = make(map[ledgercore.Txlease]basics.Round)
	}
	t.recent[rnd] = roundLeases{
		txleases: leases,
		proto:    config.Consensus[blk.CurrentProtocol],
```

Btw, I checked code and we have only read access to `txTail.txleases` so no need to allocate at all

Reply:
I like leaving it `nil` if it's read only. Seems like if we use the upsert methods properly, we ought never see places where we allocate without an immediate addition.

Reply:
Agreed left it as `nil`